app/src/main/java/com/cortado/minionify/preference/PreferencesSingleton.java

```diff
@@ -0,0 +1,46 @@
+package com.cortado.minionify.preference;
+
+import android.content.SharedPreferences;
+
+import com.cortado.minionify.mode.AppMode;
+import com.cortado.minionify.mode.exceptions.ModeNotExistsException;
+import com.cortado.minionify.preference.exceptions.PreferenceOperationFailure;
+
+public class PreferencesSingleton {
```

Comment:
no reason to use a singleton for the preferences, also there can be multiple preferences. moreover, this class has more functionality than just a singleton

Reply:
Irrelevant